tests/github-glue.test.ts

```diff
@@ -0,0 +1,204 @@
+import { Octokit } from "@octokit/rest";
+import "jest";
+import { git, gitConfig } from "../lib/git";
+import { GitHubGlue } from "../lib/github-glue";
+
+/*
+This test requires setup.  It will run successfully if setup has
+not been done.  If the test fails, there may be a pull request and a
+branch to be deleted on github and the local repo.  The test will
+attempt cleanup of a previously failed test.
+
+Setup:
+gitgitgadget.githubtest.githubuser must be configured for this test to
+identify your GitHub login.
+The value must also be configured to identify a test repo to use as
+gitgitgadget.<login>.githubrepo.
+
+Additionally, a GitHub personal access token must be set for the
+login to succeed.  This can be restricted to the test repo.
+
+For example, these configuration settings are needed (where
+`octokity` is your GitHub login):
+git config --add gitgitgadget.githubtest.githubuser octokity
+git config --add gitgitgadget.octokity.githubrepo ggg-test
+git config --add gitgitgadget.octokity.githubtoken feedbeef...
+
+The test repo must exist in github and locally.  It is expected to be
+located at the same directory level as this project.
+*/
+
+class GitHubProxy extends GitHubGlue {
+    public octo = new Octokit();
```

Comment:
Should this be initialized? The constructor always overrides it.

Taking a step back, though, I wonder whether we really have to have this class. All we want is to access a private field, right? We could always use the trick `const octo = github["client"];`

Reply:
The initializer was originally there to handle old vs current Octokit.  I had not figured out how to declare type.  Changing to specify type that makes typescript happy.

Sorry if I am missing something.  Attempts to use the private field are not working for me.

Reply:
So this does not work for you?

```js
const github = new GitHub(`../${repo}`, repo);
const octo: Octokit = github["client"];
```

Reply:
Okay, it works when I disable `no-string-literal` but that leads to other issues.  The `GitHubProxy` provides two functions.

1. It provides access to the authenticated `Octokit` api token for functions not in `GitHubGlue`.
2. It provides a method to `authenticate` using the protected `ensureAuthenticated()` method with a different owner.

The alternative to using the proxy would be to make `ensureAuthenticated()` public.

The private field could be removed as you suggested but that still leaves the authentication issue.

Reply:
Ah, okay, that makes sense!